public/app.js

```diff
@@ -0,0 +1,518 @@
+'use strict';
```

Comment:
I'm not going to go too in-depth on this file, 'cause as you've already worked out we're about to teach you some more elegant ways to write frontends. It's good to see the general MVC separation of concerns between application state/rendering/event handling functions.